src/clr-angular/forms/datepicker/date-container.ts

```diff
@@ -129,9 +133,13 @@ export class ClrDateContainer implements DynamicWrapper, OnDestroy {
     );
   }
 
-  @HostListener('body:keyup.escape')
-  close(): void {
-    this.toggleButton.nativeElement.focus();
+  @HostListener('keydown', ['$event'])
```

Comment:
Lets get rid of this listener and just react to the ifOpenService.openChange subscription. The subscribe may need to move to afterViewInit but it could look like this:

```
this._ifOpenService.openChange.subscription(open => {
  if(open) {
    this.initializeCalendar();
  } else {
    this.toggleButton.nativeElement.focus()
  }
})
```

This way we can get rid of the host listener and still react to open/close changes as needed because the host listener wasn't actually closing the datepicker conteiner.

Reply:
We could remove but there is one side effect - it's when you click outside the calendar we gonna get back the focus to the calendar icon. So if this is ok... it's a lot simple to just re-use the `if-else` block.

The above change also will remove the need for zones.

I gonna push PR with that change and we don't like it we could always revert the last commit